[C10D] Disarm PGNCCL Heartbeat Monitor to gather data

wconstab opened the pull request:
Summary:
Leave monitoring thread 'running' in log-only mode. Use the kill logs to
correlate with actual job outcomes (e.g. does stuck job detector agree?)

Later, re-enable (using a justknobs knob this time)

Test Plan: CI

Differential Revision: D53108142




cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @yf225

shuqiangzhang commented on torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:
```diff
@@ -1340,17 +1343,19 @@ void ProcessGroupNCCL::heartbeatMonitor() {
     // url here.
     const auto finalExitMsg = c10::str(logPrefix(), exitMsg);
-    terminateProcess(finalExitMsg);
+    if(monitorThreadEnabled_.load()) {
```
space after if

malfet replied:
```suggestion
    if (monitorThreadEnabled_.load()) {
```

wconstab committed:
[C10D] Disarm PGNCCL Heartbeat Monitor to gather data (#118344)

Summary:

Leave monitoring thread 'running' in log-only mode. Use the kill logs to
correlate with actual job outcomes (e.g. does stuck job detector agree?)

Later, re-enable (using a justknobs knob this time)

Test Plan: CI

Reviewed By: zdevito, shuqiangzhang, malfet, kwen2501

Differential Revision: D53108142
